src/dist/component/package.rs

```diff
@@ -228,8 +229,21 @@ fn unpack_without_first_dir<R: Read>(archive: &mut tar::Archive<R>, path: &Path)
 
         // Create the full path to the entry if it does not exist already
         match full_path.parent() {
-            Some(parent) if !parent.exists() => {
-                ::std::fs::create_dir_all(&parent).chain_err(|| ErrorKind::ExtractingPackage)?
+            Some(parent) => {
+                if !checked_parents.contains(parent) {
+                    checked_parents.insert(parent.clone().to_owned());
+                    // It would be nice to optimise this stat out, but the tar could be like so:
```

Comment:
One moderately neat way to do that would be to re-use the components iterator from above, joining and checking as you go; but this is sufficient for now

Reply:
That would result in stating every element from the top down, which is more than stating from the bottom up.

Reply:
(Or I misunderstood :). Thanks for merging it!)